packages/contentful/migrations/crn/users/20231011113618-add-validation-to-links.js

```diff
@@ -0,0 +1,148 @@
+    },
+    {
+      regexp: {
+        pattern:
+          '^(ftp|http|https):\\/\\/(\\w+:{0,1}\\w*@)?(\\S+)(:[0-9]+)?(\\/|\\/([\\w#!:.?+=&%@!\\-/]))?$',
+        flags: null,
+      },
+      message:
+        'Please enter a valid ORCID URL. (e.g. https://orcid.org/0000-000X-XXXX-XXXX)',
+    },
+  ]);
+
+  users.editField('researchGate').validations([
+    {
+      regexp: {
+        pattern:
+          '^(ftp|http|https):\\/\\/(\\w+:{0,1}\\w*@)?(\\S+)(:[0-9]+)?(\\/|\\/([\\w#!:.?+=&%@!\\-/]))?$',
+        flags: null,
+      },
+      message:
+        'Please enter a valid ResearchGate URL. (e.g. https://www.researchgate.net/profile/addyourname)',
+    },
+  ]);
+
+  users.editField('researcherId').validations([
+    {
+      regexp: {
+        pattern:
+          '^(ftp|http|https):\\/\\/(\\w+:{0,1}\\w*@)?(\\S+)(:[0-9]+)?(\\/|\\/([\\w#!:.?+=&%@!\\-/]))?$',
+        flags: null,
+        // message: 'This is the custom error 1',
+      },
+      message: 'This is the custom error 2',
+    },
+  ]);
+
+  users.changeFieldControl('linkedIn', 'builtin', 'singleLine', {
+    helpText: 'Please add the users full LinkedIn URL',
+  });
+  users.changeFieldControl('twitter', 'builtin', 'singleLine', {
+    helpText: 'Please add the users full Twitter URL',
+  });
+  users.changeFieldControl('github', 'builtin', 'singleLine', {
+    helpText: 'Please add the users full Github URL',
+  });
+  users.changeFieldControl('googleScholar', 'builtin', 'singleLine', {
+    helpText: 'Please add the users full Google Scholar URL',
+  });
+  users.changeFieldControl('orcid', 'builtin', 'singleLine', {
+    helpText: 'Please add the users full ORCID',
+  });
+  users.changeFieldControl('researchGate', 'builtin', 'singleLine', {
+    helpText: 'Please add the users full Research Gate URL',
+  });
+};
+
+module.exports.down = (migration) => {
+  const users = migration.editContentType('users');
+
+  users.editField('linkedIn').validations([]);
+  users.changeFieldControl('linkedIn', 'builtin', 'singleLine', {
+    helpText: '',
+  });
+  users.editField('twitter').validations([]);
+  users.changeFieldControl('twitter', 'builtin', 'singleLine', {
+    helpText: '',
+  });
+  users.editField('github').validations([]);
+  users.changeFieldControl('github', 'builtin', 'singleLine', {
+    helpText: '',
+  });
+  users.editField('googleScholar').validations([]);
+  users.changeFieldControl('googleScholar', 'builtin', 'singleLine', {
+    helpText: '',
+  });
+  users.editField('orcid').validations([
+    {
+      unique: true,
+    },
+  ]);
+  users.changeFieldControl('orcid', 'builtin', 'singleLine', {
+    helpText: '',
+  });
+  users.editField('researchGate').validations([]);
+  users.changeFieldControl('researchGate', 'builtin', 'singleLine', {
+    helpText: '',
+  });
+  users.editField('researcherId').validations([]);
+
+  users.changeFieldControl('orcid', 'builtin', 'singleLine', {});
+  users.changeFieldControl('researchGate', 'builtin', 'singleLine', {});
```

Comment:
these seems like duplicated calls

Reply:
So you reset set the `orcid` field on line 136 and `researchGate` field on line 140. Then you repeat the process on line 145 & 146